### PR TITLE
fix: deselect the console and silverlight transformer by default to avoid conflicts with dotnet webapp transformer

### PR DESCRIPTION
**Pull request by HarikrishnanBalagopal**
Signed-off-by: Harikrishnan Balagopal <harikrishmenon@gmail.com>

### CODE REVIEW

**Comment by HarikrishnanBalagopal** [comment]
Previous behaviour:

m2k.plan
```yaml
apiVersion: move2kube.konveyor.io/v1alpha1
kind: Plan
metadata:
  name: myproject
spec:
  sourceDir: src
  services:
    eshop:
      - transformerName: WinConsoleApp-Dockerfile
        paths:
          AppConfigFilePathList:
            - eShop.Repository.EntityFramework/App.config
          ServiceDirectories:
            - .
        configs:
          OriginalName:
            originalName: eShop
      - transformerName: WinWebApp-Dockerfile
        name: eshop
        type: Service
        paths:
          DotNetCoreCsprojPathType:
            - WebMVC/eShop.Web.csproj
            - WebAPI/eShop.WebApi.csproj
          DotNetCoreSolutionPathType:
            - eShop.sln
          ServiceDirectories:
            - .
          ServiceRootDirectory:
            - .
        configs:
          DotNet:
            isDotNetCore: false
            dotNetAppName: eShop
            isSolutionFilePresent: true
            childProjects:
              - name: eshop-web
                originalName: eShop.Web
                csProjPath: WebMVC/eShop.Web.csproj
                targetFramework: ""
              - name: eshop-webapi
                originalName: eShop.WebApi
                csProjPath: WebAPI/eShop.WebApi.csproj
                targetFramework: ""
  transformers:
    ArgoCD: m2kassets/built-in/transformers/kubernetes/argocd/transformer.yaml
    Buildconfig: m2kassets/built-in/transformers/kubernetes/buildconfig/transformer.yaml
    CloudFoundry: m2kassets/built-in/transformers/cloudfoundry/transformer.yaml
    ClusterSelector: m2kassets/built-in/transformers/kubernetes/clusterselector/transformer.yaml
    ComposeAnalyser: m2kassets/built-in/transformers/compose/composeanalyser/transformer.yaml
    ComposeGenerator: m2kassets/built-in/transformers/compose/composegenerator/transformer.yaml
    ContainerImagesPushScriptGenerator: m2kassets/built-in/transformers/containerimagespushscript/transformer.yaml
    DockerfileDetector: m2kassets/built-in/transformers/dockerfile/dockerfiledetector/transformer.yaml
    DockerfileImageBuildScript: m2kassets/built-in/transformers/dockerfile/dockerimagebuildscript/transformer.yaml
    DockerfileParser: m2kassets/built-in/transformers/dockerfile/dockerfileparser/transformer.yaml
    DotNetCore-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/dotnetcore/transformer.yaml
    EarAnalyser: m2kassets/built-in/transformers/dockerfilegenerator/java/earanalyser/transformer.yaml
    EarRouter: m2kassets/built-in/transformers/dockerfilegenerator/java/earrouter/transformer.yaml
    Golang-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/golang/transformer.yaml
    Gradle: m2kassets/built-in/transformers/dockerfilegenerator/java/gradle/transformer.yaml
    Jar: m2kassets/built-in/transformers/dockerfilegenerator/java/jar/transformer.yaml
    Jboss: m2kassets/built-in/transformers/dockerfilegenerator/java/jboss/transformer.yaml
    Knative: m2kassets/built-in/transformers/kubernetes/knative/transformer.yaml
    Kubernetes: m2kassets/built-in/transformers/kubernetes/kubernetes/transformer.yaml
    KubernetesVersionChanger: m2kassets/built-in/transformers/kubernetes/kubernetesversionchanger/transformer.yaml
    Liberty: m2kassets/built-in/transformers/dockerfilegenerator/java/liberty/transformer.yaml
    Maven: m2kassets/built-in/transformers/dockerfilegenerator/java/maven/transformer.yaml
    Nodejs-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/nodejs/transformer.yaml
    PHP-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/php/transformer.yaml
    Parameterizer: m2kassets/built-in/transformers/kubernetes/parameterizer/transformer.yaml
    Python-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/python/transformer.yaml
    ReadMeGenerator: m2kassets/built-in/transformers/readmegenerator/transformer.yaml
    Ruby-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/ruby/transformer.yaml
    Rust-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/rust/transformer.yaml
    Tekton: m2kassets/built-in/transformers/kubernetes/tekton/transformer.yaml
    Tomcat: m2kassets/built-in/transformers/dockerfilegenerator/java/tomcat/transformer.yaml
    WarAnalyser: m2kassets/built-in/transformers/dockerfilegenerator/java/waranalyser/transformer.yaml
    WarRouter: m2kassets/built-in/transformers/dockerfilegenerator/java/warrouter/transformer.yaml
    WinConsoleApp-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/windows/winconsole/transformer.yaml
    WinSLWebApp-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/windows/winsilverlightweb/transformer.yaml
    WinWebApp-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/windows/winweb/transformer.yaml
    ZuulAnalyser: m2kassets/built-in/transformers/dockerfilegenerator/java/zuul/transformer.yaml
```

**Comment by HarikrishnanBalagopal** [comment]
New behaviour:

m2k.plan
```yaml
apiVersion: move2kube.konveyor.io/v1alpha1
kind: Plan
metadata:
  name: myproject
spec:
  sourceDir: src
  services:
    eshop:
      - transformerName: WinWebApp-Dockerfile
        name: eshop
        type: Service
        paths:
          DotNetCoreCsprojPathType:
            - WebMVC/eShop.Web.csproj
            - WebAPI/eShop.WebApi.csproj
          DotNetCoreSolutionPathType:
            - eShop.sln
          ServiceDirectories:
            - .
          ServiceRootDirectory:
            - .
        configs:
          DotNet:
            isDotNetCore: false
            dotNetAppName: eShop
            isSolutionFilePresent: true
            childProjects:
              - name: eshop-web
                originalName: eShop.Web
                csProjPath: WebMVC/eShop.Web.csproj
                targetFramework: ""
              - name: eshop-webapi
                originalName: eShop.WebApi
                csProjPath: WebAPI/eShop.WebApi.csproj
                targetFramework: ""
  transformers:
    ArgoCD: m2kassets/built-in/transformers/kubernetes/argocd/transformer.yaml
    Buildconfig: m2kassets/built-in/transformers/kubernetes/buildconfig/transformer.yaml
    CloudFoundry: m2kassets/built-in/transformers/cloudfoundry/transformer.yaml
    ClusterSelector: m2kassets/built-in/transformers/kubernetes/clusterselector/transformer.yaml
    ComposeAnalyser: m2kassets/built-in/transformers/compose/composeanalyser/transformer.yaml
    ComposeGenerator: m2kassets/built-in/transformers/compose/composegenerator/transformer.yaml
    ContainerImagesPushScriptGenerator: m2kassets/built-in/transformers/containerimagespushscript/transformer.yaml
    DockerfileDetector: m2kassets/built-in/transformers/dockerfile/dockerfiledetector/transformer.yaml
    DockerfileImageBuildScript: m2kassets/built-in/transformers/dockerfile/dockerimagebuildscript/transformer.yaml
    DockerfileParser: m2kassets/built-in/transformers/dockerfile/dockerfileparser/transformer.yaml
    DotNetCore-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/dotnetcore/transformer.yaml
    EarAnalyser: m2kassets/built-in/transformers/dockerfilegenerator/java/earanalyser/transformer.yaml
    EarRouter: m2kassets/built-in/transformers/dockerfilegenerator/java/earrouter/transformer.yaml
    Golang-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/golang/transformer.yaml
    Gradle: m2kassets/built-in/transformers/dockerfilegenerator/java/gradle/transformer.yaml
    Jar: m2kassets/built-in/transformers/dockerfilegenerator/java/jar/transformer.yaml
    Jboss: m2kassets/built-in/transformers/dockerfilegenerator/java/jboss/transformer.yaml
    Knative: m2kassets/built-in/transformers/kubernetes/knative/transformer.yaml
    Kubernetes: m2kassets/built-in/transformers/kubernetes/kubernetes/transformer.yaml
    KubernetesVersionChanger: m2kassets/built-in/transformers/kubernetes/kubernetesversionchanger/transformer.yaml
    Liberty: m2kassets/built-in/transformers/dockerfilegenerator/java/liberty/transformer.yaml
    Maven: m2kassets/built-in/transformers/dockerfilegenerator/java/maven/transformer.yaml
    Nodejs-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/nodejs/transformer.yaml
    PHP-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/php/transformer.yaml
    Parameterizer: m2kassets/built-in/transformers/kubernetes/parameterizer/transformer.yaml
    Python-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/python/transformer.yaml
    ReadMeGenerator: m2kassets/built-in/transformers/readmegenerator/transformer.yaml
    Ruby-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/ruby/transformer.yaml
    Rust-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/rust/transformer.yaml
    Tekton: m2kassets/built-in/transformers/kubernetes/tekton/transformer.yaml
    Tomcat: m2kassets/built-in/transformers/dockerfilegenerator/java/tomcat/transformer.yaml
    WarAnalyser: m2kassets/built-in/transformers/dockerfilegenerator/java/waranalyser/transformer.yaml
    WarRouter: m2kassets/built-in/transformers/dockerfilegenerator/java/warrouter/transformer.yaml
    WinWebApp-Dockerfile: m2kassets/built-in/transformers/dockerfilegenerator/windows/winweb/transformer.yaml
    ZuulAnalyser: m2kassets/built-in/transformers/dockerfilegenerator/java/zuul/transformer.yaml
```

File: Makefile [hunk 1]
```diff
@@ -134,7 +135,11 @@ test-coverage: ${GOLANGCOVER} ## Run tests with coverage
 	go test -run . $(PKG) -coverprofile=coverage.txt -covermode=atomic
 
 ${GOLANGCILINT}:
-	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(GOPATH)/bin v1.45.2
+ifeq ($(HAS_LINT),false)
```
Comment (ashokponkumar) [inline Makefile:138]: Why are we removing this?

**Review by ashokponkumar — CHANGES_REQUESTED** [review]
Nice work.

File: Makefile (hunk 1 above)
Comment (HarikrishnanBalagopal) [reply Makefile:139]: because it fails

Comment (HarikrishnanBalagopal) [reply Makefile:139]: also they have different instructions for installing locally vs in a CI/CD pipeline.

**Review by ashokponkumar — COMMENTED** [review]
Is there a way we can show to users in the plan the list of all available transformers too. Else they have to go to code to find them or use plan+transform always which is not a good experience.

File: transformer/transformer.go [hunk 2]
```diff
@@ -176,11 +177,21 @@ func InitTransformers(transformerToInit map[string]string, selector labels.Selec
 	}
 	transformerConfigs := getFilteredTransformers(transformerToInit, selector, logError)
 	transformerNames := []string{}
-	for transformerName := range transformerConfigs {
+	defaultSelectedTransformerNames := []string{}
+	for transformerName, t := range transformerConfigs {
 		transformerNames = append(transformerNames, transformerName)
+		if v, ok := t.ObjectMeta.Labels["move2kube.konveyor.io/default-selected"]; !ok || cast.ToBool(v) {
```
Comment (ashokponkumar) [inline transformer/transformer.go:183]: Use groupname from types

**Comment by HarikrishnanBalagopal** [comment]
> Is there a way we can show to users in the plan the list of all available transformers too. Else they have to go to code to find them or use plan+transform always which is not a good experience.

The transformer is shown during the transform phase. They can enable it if they want. This change only removes transformers that have a certain label from the default answer.
They don't need to look at the code to find the transformer.

File: transformer/transformer.go (hunk 2 above)
Comment (HarikrishnanBalagopal) [reply transformer/transformer.go:183]: I don't see it in types. It should be there somewhere........

Comment (HarikrishnanBalagopal) [reply transformer/transformer.go]: done

**Comment by ashokponkumar** [comment]
> > Is there a way we can show to users in the plan the list of all available transformers too. Else they have to go to code to find them or use plan+transform always which is not a good experience.
> 
> The transformer is shown during the transform phase. They can enable it if they want. This change only removes transformers that have a certain label from the default answer.
> They don't need to look at the code to find the transformer.

If somebody runs a plan and then transform separately, then I don't think the transformers that have this flag will be listed either in the plan or in transform.

**Comment by HarikrishnanBalagopal** [comment]
> > > Is there a way we can show to users in the plan the list of all available transformers too. Else they have to go to code to find them or use plan+transform always which is not a good experience.
> > 
> > 
> > The transformer is shown during the transform phase. They can enable it if they want. This change only removes transformers that have a certain label from the default answer.
> > They don't need to look at the code to find the transformer.
> 
> If somebody runs a plan and then transform separately, then I don't think the transformers that have this flag will be listed either in the plan or in transform.

isn't that expected?

**Comment by ashokponkumar** [comment]
> > > > Is there a way we can show to users in the plan the list of all available transformers too. Else they have to go to code to find them or use plan+transform always which is not a good experience.
> > > 
> > > 
> > > The transformer is shown during the transform phase. They can enable it if they want. This change only removes transformers that have a certain label from the default answer.
> > > They don't need to look at the code to find the transformer.
> > 
> > If somebody runs a plan and then transform separately, then I don't think the transformers that have this flag will be listed either in the plan or in transform.
> 
> isn't that expected?

In that case nobody will even know what other transform options they have. It might be better if even in that scenario, if the question in transform shows all transformers with the selected transformers alone selected by default.

**Comment by HarikrishnanBalagopal** [comment]
> In that case nobody will even know what other transform options they have. It might be better if even in that scenario, if the question in transform shows all transformers with the selected transformers alone selected by default.

I don't see a reason to bother the user with a ton of new transformers that weren't in the plan.

**Comment by HarikrishnanBalagopal** [comment]
`In that case nobody will even know what other transform options they have.`

They still see it if they do the single step `move2kube transform -s src/`

**Comment by ashokponkumar** [comment]
Say, somebody wants to know whether their transformer was even detected successfully, there is no way in this flow. It makes it hard. The plan was giving them the option that it was detected.

**Comment by HarikrishnanBalagopal** [comment]
> Say, somebody wants to know whether their transformer was even detected successfully, there is no way in this flow. It makes it hard. The plan was giving them the option that it was detected.

This doesn't affect any custom transformers unless they add the label. By default a transformer is selected (even if they don't have the label).

**Comment by ashokponkumar** [comment]
> > Say, somebody wants to know whether their transformer was even detected successfully, there is no way in this flow. It makes it hard. The plan was giving them the option that it was detected.
> 
> This doesn't affect any custom transformers unless they add the label and set it to false.
> By default a transformer is selected (even if they don't have the label).

Yes, but it would be good to provide a way to list all transformers

**Pull request closed without merging**